rmgpy/species.py

```diff
@@ -519,19 +520,22 @@ def copy(self, deep=False):
     def getAugmentedInChI(self):
         if self.aug_inchi is None:
             self.aug_inchi = self.generate_aug_inchi()
-            return self.aug_inchi
-        else:
-            return self.aug_inchi
+        return self.aug_inchi
 
     def generate_aug_inchi(self):
         candidates = []
         self.generateResonanceIsomers()
         for mol in self.molecule:
-            cand = mol.toAugmentedInChI()
-            candidates.append(cand)
-
-        candidates.sort()
-        return candidates[0] 
+            try:
+                cand = [mol.toAugmentedInChI(),mol]
+                candidates.append(cand)
+            except:
+                pass  # not all resonance isomers can be parsed into InChI (e.g. if containing a hypervalance atom)
+        candidates = sorted(candidates, key=itemgetter(0))
+        for cand in candidates:
+            if not any(atom.charge != 0 for atom in cand[1].vertices):
+                return cand[0]
```

Comment:
Just to confirm, this preferentially returns InChI's of charged resonance forms?

Reply:
This should preferentially return InChI's of *uncharged* forms.
The `any()` is TRUE if one or more atoms are charged, and the `not` reverses it.

Reply:
Thanks, I missed the not...

Would this be equivalent to `all(atom.charge == 0 for atom in cand[1].vertices)`?

Reply:
Yes, thanks. I'll change it.